### DISCUSSION TITLE
How do i add attribute on signin?

### DISCUSSION THREAD

**Discussion by Glutch** [nextauthjs/next-auth#1503] [Help]
I'm using the Discord provider and want to grab the profile.descriminator so i can get the username#1234 value. This is only available in SignIn.

I've tried this 
```javascript
async signIn(user, account, profile) {
  try {
    await dbConnect()
    await new User({
      ...user,
      discordId: profile.id,
      discord: `${profile.username}#${profile.discriminator}`,
    }).save()
    return true
  } catch (e) {
    return false
  }
},
```

But i'm getting "To confirm your identity, sign in with the same account you used originally."
Url: http://localhost:3000/api/auth/signin?error=**OAuthAccountNotLinked**

How would i properly add custom values upon signin?

I assume i have to write a custom model. But there is no information i can find how to do this with mongoose. Any examples would be greatly appreciated!

**Comment by Glutch** [comment]
I'm still trying to figure this out after 3 days, any help greatly appreciated 🙏

**Comment by StefanSelfTaught** [comment]
You could probably create your custom mongoose models based on the default NextAuth models. Maybe with the mongodb node client would be a little bit easier.

Also in the `signIn` callback you should update the user document.

**Reply by Glutch** [reply]
Thank you for taking the time to try and help, i'm pretty desperate to fix this atm. As you can see i've already tried to create the user document during the signIn call (adding the entire nextauth user object by spreading it), it does "work" - the user is created in my database but i get this error: "To confirm your identity, sign in with the same account you used originally.".

And the "sessions" and "accounts" collections are **not** created in my database. (Thats probably the reason i can't login and is getting that error).

**Comment by Glutch** [comment]
I could solve this by during signup, create a separate user object called "UserInfo", and store all custom info there. But that would mean i can't access any of this info through the session, only separate api calls. Which i guess works as a workaround but i rather solve it how its meant to be solved

**Comment by StefanSelfTaught** [comment]
The `sessions` and `accounts` collections should be created. So I was saying in the `signIn` callback to add a new field on the `User` document  after it was created. You could also use a custom adapter but I don't know if you would like this.

**Reply by Glutch** [reply]
Could you provide a code example on how to add a new field to the user document? Feels like i've tried everything

**Reply by StefanSelfTaught** [reply]
Hopefully the user is created before this 😃 
```js
async signIn(user, account, profile) {
   await Model.findByIdAndUpdate(user.id, { $set: { discord: `${profile.username}#${profile.discriminator}` }})
   return true
}

**Reply by Glutch** [reply]
> Hopefully the user is created before this 😃
> 
> ```js
> async signIn(user, account, profile) {
>    await Model.findByIdAndUpdate(user.id, { $set: { discord: `${profile.username}#${profile.discriminator}` }})
>    return true
> }
> ```

The user is not created at that point :/

**Reply by StefanSelfTaught** [reply]
If you are using jwt you can do this in the jwt callback

**Reply by Glutch** [reply]
Yes but i'm using mongo atlas so i dont think theres any reason to use jwt? or is there? should you?

**Reply by StefanSelfTaught** [reply]
It depends. You can use jwt for the session instead of storing the session in the database. You can read more here: https://next-auth.js.org/faq#json-web-tokens
